### ISSUE TITLE
Remove Media files for a removed tenant when using Azure Blob Storage

### ISSUE THREAD

**Issue by Piedone** [OrchardCMS/OrchardCore#16403]
### Is your feature request related to a problem? Please describe.
If you remove a tenant, its Media files are only removed if they use the default local storage in App_Data. When using Azure Blob Storage though, they remain.

Related: https://github.com/OrchardCMS/OrchardCore/issues/4950.

### Describe the solution you'd like
Add an `IShellRemovingHandler` to `OrchardCore.Media.Azure` that removed Media files from Azure Blob Storage too. This should support both removing the whole container if tenants' Media are stored in separate containers, or just the "folder" if a single container is used.

### Describe alternatives you've considered
You can also have a custom `IShellRemovingHandler` implementation for this, but I think this should be a core functionality.

**Comment by porgabi** [comment]
What I've found is the logic to remove the container upon tenant deletion is [already there](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Media.Azure/Services/MediaBlobContainerTenantEvents.cs#L57), but it's not run because the corresponding settings are bugged (i.e. setting `RemoveContainer` to `true` in appsettings does not actually take effect). With a fix in place for that, the container is indeed deleted along with the tenant.

> This should support both removing the whole container if tenants' Media are stored in separate containers, or just the "folder" if a single container is used.

To account for this, the existing settings could be extended with an additional flag (e.g. `RemoveFilesFromBasePath`), which could then determine whether to delete the container or only the files that are associated with the given base path. So, if I understand correctly, there's no need for a new handler here, only have to extend settings and the logic within `RemovingAsync()` in the above linked file.

**Comment by Piedone** [comment]
Yes, indeed, this just needs to be extended by the folder deletion, and a config for that.
